x-pack/plugins/search_playground/server/routes.ts

```diff
@@ -45,22 +61,42 @@ export function defineRoutes({ log, router }: { log: Logger; router: IRouter })
       path: APIRoutes.POST_CHAT_MESSAGE,
       validate: {
         body: schema.object({
-          data: schema.any(),
+          data: schema.object({
+            connector_id: schema.string(),
+            indices: schema.string(),
+            prompt: schema.string(),
+            citations: schema.boolean(),
+            elasticsearch_query: schema.string(),
+            summarization_model: schema.string(),
+            doc_size: schema.number(),
+            source_fields: schema.string(),
+          }),
           messages: schema.any(),
         }),
       },
     },
     errorHandler(async (context, request, response) => {
+      const [_, { encryptedSavedObjects }] = await getStartServices();
       const { client } = (await context.core).elasticsearch;
+      const encryptedSavedObjectsClient = encryptedSavedObjects.getClient({
```

Comment:
Rather than get the connector's API key by reading the ESO directly, and then calling OpenAPI functions directly, you should be using the ActionClient to execute the connector, like other gen-ai based route handlers do.

Unless for some reason you can't.  In which case we'll need to dig further.

Basically, by doing this, a lot of validation code is bypassed, so it's possible folks could be using the connector in ways that we've otherwise not allowed.  A passive example of that we won't get any audit documents for these invocations, but we would if the actions client was used.

I'd suggest contacting @stephmilovic on how to do this ...

Reply:
This PR was changed to use the existing connector directly, resulting in just one change to ResponseOps-owned code - the module that defines the new connector feature.